actionpack/lib/action_controller/metal/params_wrapper.rb

```diff
@@ -253,10 +253,9 @@ def process_action(*args)
         # This will display the wrapped hash in the log file.
         request.filtered_parameters.merge! wrapped_filtered_hash
       end
-    ensure
-      # NOTE: Rescues all exceptions so they
-      # may be caught in ActionController::Rescue.
-      return super
+      super
+    rescue ActionDispatch::Http::Parameters::ParseError
+      super
```

Comment:
Should `super` be called again? If it raised, it will raise again.

Reply:
I guess we want to rescue this exception only the in the pace that raises it, so we know what is happening. I think it is the calls inside the `if`

Reply:
Correct. I think trying to move the existing logic in the `if` above into a private method, and then rescuing parse errors there would work better.

Reply:
Moved into a private method, I have called it `wrap` but I think it could do with a better method name? 

Was not sure I understood when rescuing the parse error, do we still want to cool `super` or swallow the exception? Please have a look at updated code and let me know.

Reply:
I think `wrap_parameters` would be a suitable name. Swallowing the exception here is preferred because it will call `super` immediately afterward in `process_action`.

Reply:
@gmcgibbon would that be confusing with the class method `wrap_parameters` and the internal instance `_wrap_parameters`?

Reply:
Yeah that's the reason why I never named it `wrap_parameters` too... @gmcgibbon any other name you prefer?

`_perform_wrap` 
`parameter_wrapping`
`_perform_parameter_wrapping`

Reply:
My mistake. I suppose `_perform_parameter_wrapping` is better in that case.

Reply:
Cool, done and squashed the commits too. 👍